include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -95,6 +95,29 @@ class pool_memory_resource final
 
     initialize_pool(initial_pool_size, maximum_pool_size);
   }
+  
+  /**
+   * @brief Construct a `pool_memory_resource` and allocate the initial device memory pool using
+   * `upstream_mr`.
+   *
+   * @throws rmm::logic_error if `upstream_mr == nullptr`
+   * @throws rmm::logic_error if `initial_pool_size` is neither the default nor aligned to a
+   * multiple of pool_memory_resource::allocation_alignment bytes.
+   * @throws rmm::logic_error if `maximum_pool_size` is neither the default nor aligned to a
+   * multiple of pool_memory_resource::allocation_alignment bytes.
+   *
+   * @param upstream_mr The memory_resource from which to allocate blocks for the pool.
+   * @param initial_pool_size Minimum size, in bytes, of the initial pool. Defaults to half of the
+   * available memory on the current device.
+   * @param maximum_pool_size Maximum size, in bytes, that the pool can grow to. Defaults to all
+   * of the available memory on the current device.
+   */
+  template <typename Upstream2 = Upstream, 
+            cuda::std::enable_if_t<cuda::mr::resource_with<Upstream, cuda::mr::device_accessible>, int> = 0>
```

Comment:
```suggestion
            cuda::std::enable_if_t<cuda::mr::resource<Upstream>, int> = 0>
```

Reply:
Let's also add a test that uses the `rmm::pinned_memory_resource` as the upstream to the `pool_memory_resource`.

Reply:
I just realized that this wont properly work. Currently we derive `rmm::pool_memory_resource` from `stream_ordered_resource`, which is a `cuda::device_accessible` resource

Reply:
Can we do this now that you've added the `maybe_remove_property` thing?

Reply:
Oh sorry, I already did and this is just a stale comment. Closing 

Note: I did change it to `cuda::mr::async_resource` because that is what we actually want

Reply:
What I meant was we should add a test for `pool_memory_resource` using `pinned_memory_resource` as the upstream.

Reply:
🤦‍♂️ will do that tomorrow, as obviously it is too late today